eclipse-collections-code-generator/src/main/resources/api/map/objectPrimitiveMap.stg

```diff
@@ -94,6 +94,31 @@ public interface Object<name>Map\<K> extends <name>Iterable
     void forEachKeyValue(Object<name>Procedure\<? super K> procedure);
     <if(!primitive.booleanPrimitive)><(flipUniqueValues.(name))(name)><endif>
 
+    /**
+     * This method can be used to implement anySatisfy, allSatisfy, and noneSatisfy.
+     *
+     * anySatisfy  = this.shortCircuitKeyValue(predicate, true, true, false);
+     * allSatisfy  = this.shortCircuitKeyValue(predicate, false, false, true);
+     * noneSatisfy = this.shortCircuitKeyValue(predicate, true, false, true);
+     *
+     * @since 12.0
+     */
+    default boolean shortCircuitKeyValue(
+            Object<name>Predicate\<K> predicate,
+            boolean shortCircuitCondition,
```

Comment:
at first blush, this is a rather odd function signature. the inclusion of `shortCircuitCondition` goes against the grain of having a `Predicate` -- I don't think there is anywhere else where we compare the output of a predicate with a parameter. Normally, we just expect the predicate to return true for whatever condition the function requires, which necessitates the use of `Predicates.not` on occasion.

I imagine removing this parameter would slow down `allSatisfy`, but it would be worth measuring that.

Reply:
I probably should have marked this PR as draft. I originally had all three implementations on the api (`any`, `all`, `none`) delegating to the shortCircuit method which I borrowed from `UnifiedMap`. I removed all of them from the PR to make it easier to review. The only one that is needed right now is allSatisfy, which I could add back. I thought it would be better to have a separate PR for all three methods with tests, and only including what was necessary for the performance improvement in this one. I think I'll just go with `allSatisfyKeyValue` on the API and get rid of the shortCircuit method. We can always add `any` and `none` later.